instance/tests/models/test_openedx_database_mixins.py

```diff
@@ -0,0 +1,285 @@
+            self.instance.provision_mongo()
+            databases = mongo.database_names()
+            for database in self.instance.mongo_database_names:
+                self.assertNotIn(database, databases)
+
+    def test_provision_mongo_again(self):
+        """
+        Only create the databases once
+        """
+        self.instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        self.instance.provision_mongo()
+        self.assertIs(self.instance.mongo_provisioned, True)
+
+        mongo_user = self.instance.mongo_user
+        mongo_pass = self.instance.mongo_pass
+        self.instance.provision_mongo()
+        self.assertEqual(self.instance.mongo_user, mongo_user)
+        self.assertEqual(self.instance.mongo_pass, mongo_pass)
+        self.check_mongo()
+
+    @override_settings(INSTANCE_MONGO_URL_OBJ=urlparse('mongodb://user:pass@mongo.opencraft.com'))
+    def test_ansible_settings_mongo(self):
+        """
+        Add mongo ansible vars if INSTANCE_MONGO_URL is set
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        appserver = make_test_appserver(instance)
+        ansible_vars = appserver.configuration_settings
+        self.assertIn('EDXAPP_MONGO_USER: {0}'.format(instance.mongo_user), ansible_vars)
+        self.assertIn('EDXAPP_MONGO_PASSWORD: {0}'.format(instance.mongo_pass), ansible_vars)
+        self.assertIn('EDXAPP_MONGO_HOSTS: [mongo.opencraft.com]', ansible_vars)
+        self.assertIn('EDXAPP_MONGO_PORT: 27017', ansible_vars)
+        self.assertIn('EDXAPP_MONGO_DB_NAME: {0}'.format(instance.mongo_database_name), ansible_vars)
+        self.assertIn('FORUM_MONGO_USER: {0}'.format(instance.mongo_user), ansible_vars)
+        self.assertIn('FORUM_MONGO_PASSWORD: {0}'.format(instance.mongo_pass), ansible_vars)
+        self.assertIn('FORUM_MONGO_HOSTS: [mongo.opencraft.com]', ansible_vars)
+        self.assertIn('FORUM_MONGO_PORT: 27017', ansible_vars)
+        self.assertIn('FORUM_MONGO_DATABASE: {0}'.format(instance.forum_database_name), ansible_vars)
+
+    @override_settings(INSTANCE_MONGO_URL_OBJ=None)
+    def test_ansible_settings_mongo_not_set(self):
+        """
+        Don't add mongo ansible vars if INSTANCE_MONGO_URL is not set
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=True)
+        appserver = make_test_appserver(instance)
+        self.check_mongo_vars_not_set(appserver)
+
+    @override_settings(INSTANCE_MONGO_URL_OBJ=urlparse('mongodb://user:pass@mongo.opencraft.com'))
+    def test_ansible_settings_mongo_ephemeral(self):
+        """
+        Don't add mongo ansible vars if INSTANCE_MONGO_URL is not set
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=True)
+        appserver = make_test_appserver(instance)
+        self.check_mongo_vars_not_set(appserver)
+
+    def check_mongo_vars_not_set(self, appserver):
```

Comment:
@bradenmacdonald Nit: Move above `test_*` methods (below `check_mongo`) to keep helper methods and test methods separate.